pp/go/relabeler/head/catalog/log_migrate.go

```diff
@@ -0,0 +1,147 @@
+package catalog
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/prometheus/prometheus/pp/go/relabeler/logger"
+)
+
+var (
+	ErrUnsupportedVersion = errors.New("unsupported version")
+	ErrUnreadableLogFile  = errors.New("unreadable log file")
+)
+
+const (
+	headerSize = 8
+)
+
+func migrate(filePath string, targetVersion uint64) (_ *FileHandler, _ Encoder, _ Decoder, _ error) {
+	var forceReWrite bool
+	file, version, encoder, decoder, err := loadFile(filePath)
+	if err != nil {
+		if !errors.Is(err, ErrUnreadableLogFile) {
+			return nil, nil, nil, err
+		}
+
+		forceReWrite = true
+		compactedFilePath := fmt.Sprintf("%s.compacted", filePath)
+		file, version, encoder, decoder, err = loadFile(compactedFilePath)
```

Comment:
I think we should simplify this block to next steps:
1. Try read catalog
2. If there is no catalog, check is there compacted catalog and back to step 1.